src/components/ColumnMenu/ColumnMenu.tsx

```diff
@@ -117,8 +126,42 @@ export default function ColumnMenu() {
   const userDocHiddenFields =
     userSettings.tables?.[formatSubTableName(tableId)]?.hiddenFields ?? [];
 
+  let referencedColumns: string[] = [];
+  let referencedExtensions: string[] = [];
+  Object.entries(tableSchema?.columns ?? {}).forEach(([key, c], index) => {
+    if (
+      c.config?.listenerFields?.includes(column.key) ||
+      c.config?.requiredFields?.includes(column.key)
+    ) {
+      referencedColumns.push(key);
```

Comment:
```suggestion
      referencedColumns.push(c.name);
```
Display user-facing column name instead